modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java

```diff
@@ -106,101 +119,140 @@ public ReplicaManager(
+                        return replicaFut;
+                    }
+                });
 
-                            return replicaFut;
-                        }
-                    });
+                return;
+            }
 
-                    return;
-                }
+            CompletableFuture<Replica> replicaFut = replicas.get(request.groupId());
 
-                CompletableFuture<Replica> replicaFut = replicas.get(request.groupId());
+            HybridTimestamp requestTimestamp = extractTimestamp(request);
 
-                HybridTimestamp requestTimestamp = extractTimestamp(request);
+            if (replicaFut == null || !replicaFut.isDone()) {
+                sendReplicaUnavailableErrorResponse(senderConsistentId, correlationId, request.groupId(), requestTimestamp);
 
-                if (replicaFut == null || !replicaFut.isDone()) {
-                    sendReplicaUnavailableErrorResponse(senderConsistentId, correlationId, request, requestTimestamp);
+                return;
+            }
 
-                    return;
-                }
+            // replicaFut is always completed here.
+            CompletableFuture<?> result = replicaFut.join().processRequest(request);
 
-                // replicaFut is always completed here.
-                CompletableFuture<?> result = replicaFut.join().processRequest(request);
+            result.handle((res, ex) -> {
+                NetworkMessage msg;
 
-                result.handle((res, ex) -> {
-                    NetworkMessage msg;
+                if (ex == null) {
+                    msg = prepareReplicaResponse(requestTimestamp, res);
+                } else {
+                    LOG.warn("Failed to process replica request [request={}]", ex, request);
 
-                    if (ex == null) {
-                        msg = prepareReplicaResponse(requestTimestamp, res);
-                    } else {
-                        LOG.warn("Failed to process replica request [request={}]", ex, request);
+                    msg = prepareReplicaErrorResponse(requestTimestamp, ex);
+                }
 
-                        msg = prepareReplicaErrorResponse(requestTimestamp, ex);
-                    }
+                clusterNetSvc.messagingService().respond(senderConsistentId, msg, correlationId);
 
-                    clusterNetSvc.messagingService().respond(senderConsistentId, msg, correlationId);
+                return null;
+            });
+        } finally {
+            busyLock.leaveBusy();
+        }
+    }
 
-                    return null;
-                });
-            } finally {
-                busyLock.leaveBusy();
+    private void onPlacementDriverMessageReceived(NetworkMessage msg0, String senderConsistentId, @Nullable Long correlationId) {
+        if (!busyLock.enterBusy()) {
+            throw new IgniteException(new NodeStoppingException());
+        }
+
+        try {
+            if (!(msg0 instanceof PlacementDriverReplicaMessage)) {
```

Comment:
Which other types of messages do you wanting here? I doubt the other type is possible.
It this case, assert is more preferable.